examples/tensorflow/image-classification/run_image_classification.py

```diff
@@ -0,0 +1,567 @@
+
+    model_name_or_path: str = field(
+        default="google/vit-base-patch16-224-in21k",
+        metadata={"help": "Path to pretrained model or model identifier from huggingface.co/models"},
+    )
+    model_type: Optional[str] = field(
+        default=None,
+        metadata={"help": "If training from scratch, pass a model type from the list: " + ", ".join(MODEL_TYPES)},
+    )
+    config_name: Optional[str] = field(
+        default=None, metadata={"help": "Pretrained config name or path if not the same as model_name"}
+    )
+    cache_dir: Optional[str] = field(
+        default=None, metadata={"help": "Where do you want to store the pretrained models downloaded from s3"}
+    )
+    model_revision: str = field(
+        default="main",
+        metadata={"help": "The specific model version to use (can be a branch name, tag name or commit id)."},
+    )
+    feature_extractor_name: str = field(default=None, metadata={"help": "Name or path of preprocessor config."})
+    use_auth_token: bool = field(
+        default=False,
+        metadata={
+            "help": (
+                "Will use the token generated when running `huggingface-cli login` (necessary to use this script "
+                "with private models)."
+            )
+        },
+    )
+    ignore_mismatched_sizes: bool = field(
+        default=False,
+        metadata={"help": "Will enable to load a pretrained model whose head dimensions are different."},
+    )
+
+
+def center_crop(image, size):
+    size = (size, size) if isinstance(size, int) else size
+    orig_height, orig_width, _ = image.shape
+    crop_height, crop_width = size
+    top = (orig_height - orig_width) // 2
+    left = (orig_width - crop_width) // 2
+    image = tf.image.crop_to_bounding_box(image, top, left, crop_height, crop_width)
+    return image
+
+
+# Numpy and TensorFlow compatible version of PyTorch RandomResizedCrop. Code adapted from:
+# https://pytorch.org/vision/main/_modules/torchvision/transforms/transforms.html#RandomResizedCrop
+def random_crop(image, scale=(0.08, 1.0), ratio=(3.0 / 4.0, 4.0 / 3.0)):
+    height, width, _ = image.shape
+    area = height * width
+    log_ratio = np.log(ratio)
+    for _ in range(10):
+        target_area = np.random.uniform(*scale) * area
+        aspect_ratio = np.exp(np.random.uniform(*log_ratio))
+        w = int(round(np.sqrt(target_area * aspect_ratio)))
+        h = int(round(np.sqrt(target_area / aspect_ratio)))
+        if 0 < w <= width and 0 < h <= height:
+            i = np.random.randint(0, height - h + 1)
+            j = np.random.randint(0, width - w + 1)
+            return image[i : i + h, j : j + w, :]
+
+    # Fallback to central crop
+    in_ratio = float(width) / float(height)
+    w = width if in_ratio < min(ratio) else int(round(height * max(ratio)))
+    h = height if in_ratio > max(ratio) else int(round(width / min(ratio)))
+    i = (height - h) // 2
+    j = (width - w) // 2
+    return image[i : i + h, j : j + w, :]
```

Comment:
I know this is from torchvision and not your idea, but trying 10 random rolls and then giving up rather than figuring out how to constrain your randomness so it outputs valid solutions the first time is quite funny.

Reply:
Although on reflection I suppose doing it this way gives you an unbiased sample from the space of valid target_area + aspect ratio pairs, and you'd need to handle things quite carefully to make that the case if you constrained the random rolls to always be valid.